src/main/java/io/github/nihaljain/JettyServer9.java

```diff
@@ -15,14 +15,14 @@ public class JettyServer9 {
   private static final Logger logger = LoggerFactory.getLogger(JettyServer9.class);
 
   public static void main(String[] args) throws Exception {
-    String logDirPath = Util.getLogDirPath(args);
+    Path logDirPath = Util.getLogDirPath(args);
 
     Server server = new Server(8080);
     ContextHandlerCollection parent = new ContextHandlerCollection();
 
     ServletContextHandler logContext = new ServletContextHandler(parent, "/logs");
     logContext.addServlet(new ServletHolder(new DefaultServlet()), "/");
-    logContext.setResourceBase(logDirPath);
+    logContext.setResourceBase(logDirPath.toString());
```

Comment:
```suggestion
    logContext.setBaseResourceAsPath(logDirPath);
```

You are still using the deprecated method, use the non-deprecated option.

Reply:
This file maps to Jetty 9 and uses jetty 9 to compile and hence would fail with
```
[ERROR] /Users/nihjain/code/personal/jetty_log_dir_issue/src/main/java/io/github/nihaljain/JettyServer9.java:[25,15] cannot find symbol
[ERROR]   symbol:   method setBaseResourceAsPath(java.nio.file.Path)
[ERROR]   location: variable logContext of type org.eclipse.jetty.servlet.ServletContextHandler
```
 The new method applies to only Jetty12 right. Hence I had updated the other file i.e. JettyServer12.java only.

Reply:
Gotcha.

Reply:
```suggestion
    logContext.setResourceBase(logDirPath.toUri().toASCIIString());
```
Then use the URI string reference instead for Jetty 9.

Reply:
Done!